AsyncDisplayKit/ASVideoNode.h

```diff
@@ -15,7 +15,7 @@
 // in an issue on GitHub: https://github.com/facebook/AsyncDisplayKit/issues
 
 @interface ASVideoNode : ASControlNode
-@property (atomic, strong, readwrite) AVAsset *asset;
+@property (atomic, strong, readonly) AVAsset *asset;
```

Comment:
Done, gone back to the property init method and already had asserts to ensure either URL or AVAsset construction. Seems to work fine.